### PR TITLE
add version prompt and bump JSON to 0.2 (#185)

### LINKED ISSUE #185: Version of the JSON format used in project file
Right now we may expect some changes in the JSON file generated by UI (to be consumed by backend).
Some changes are just extensions, like https://github.com/yaptide/ui/pull/182 , but more changes in the future not be backward compatible.
To deal with such issues in the converter code it would be nice to have some versioning of the JSON schema.
Can we add such a version number to the top-level structure in JSON ?

I see that threejs have some versioning number for their own files, like documented here:
https://github.com/mrdoob/three.js/wiki/JSON-Object-Scene-format-4
It seems that now they have v 4.3 ?

### PR DESCRIPTION
**Pull request by ostatni5**
close #185

### CODE REVIEW

File: src/ThreeEditor/js/Loader.js [hunk 1]
```diff
@@ -155,7 +155,17 @@ function Loader(editor) {
 				break;
 
 			case 'editor':
-				if (window.confirm('Current editor data will be lost. Are you sure?')) {
+				let versionIsOk = true;
+				if (data.metadata.version !== editor.jsonVersion) {
+					versionIsOk = window.confirm(
+						`File was generated with different version of editor JSON: ${data.metadata.version}\nCurrent version of editor JSON: ${editor.jsonVersion}\nContinue?`
```
Comment (grzanka) [inline src/ThreeEditor/js/Loader.js:161]: What about backward and forward compatibility ? Maybe we should use some proper versioning compare methods ?

Comment (grzanka) [reply src/ThreeEditor/js/Loader.js:161]:
```suggestion
						`File has  project standard version ${data.metadata.version} which is different from standard version handled by this web application: ${editor.jsonVersion}\nContinue?`
```

Comment (ostatni5) [reply src/ThreeEditor/js/Loader.js]: > 

I think we should care about backward compatibility, after the first full release(end of engineer thesis ). And then if JSON will change we could add methods to handle old versions.  

More validation of deserialized JSON and default values when loading data could help with compatibility, but it is another issue.

File: src/ThreeEditor/main.js [hunk 2]
```diff
@@ -48,7 +49,14 @@ export function initEditor(container) {
 			if (isLoadingFromHash) return;
 
 			if (typeof state !== 'undefined') {
-				editor.fromJSON(state);
+				let versionIsOk = true;
+				if (state.metadata.version !== editor.jsonVersion) {
+					versionIsOk = window.confirm(
+						`Editor in memory has version of JSON: ${state.metadata.version}\nCurrent version of editor JSON: ${editor.jsonVersion}\nLoad it anyway?`
```
Comment (grzanka) [inline src/ThreeEditor/main.js:55]: Shouldn't this be consistent with:

image

?

Comment (ostatni5) [reply src/ThreeEditor/main.js:55]: One message is for loading form file one for loading from browser database.

Comment (grzanka) [reply src/ThreeEditor/main.js:55]: OK

**Review by grzanka — APPROVED** [review]
LGTM

**Pull request merged**